### PR TITLE
Onboarding Improvements: Epilogue screen UI

### PR DESCRIPTION
**Pull request by nikola-milicevic**
Part of #17385

This PR includes redesigned UI for Login Epilogue screen.

## Note
- Making the sites tappable / hooking up the create a new site button will be addressed in a subsequent PR
- Fixing the failing UI tests will be addressed in a subsequent PR

### Test variation 1
1. Log in with user that has less than 4 sites
2. Confirm UI matches new design

### Test variation 2
1. Log in with user that has more than 4 sites
2. Confirm UI matches new design

### Screenshots
App | Variation 1 | Variation 2
-- | -- | --
WordPress |  | 
Jetpack |  | 

### Regression Notes
1. Potential unintended areas of impact
Jetpack -- The Create A New Site button should be hidden

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested the above steps, see the screenshots

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/LoginEpilogueChooseSiteTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    func setupSubtitleLabel() {
+        subtitleLabel.text = NSLocalizedString("You can switch sites at any time.", comment: "A text for subtitle label on Login epilogue screen")
+        subtitleLabel.font = WPStyleGuide.fontForTextStyle(.footnote, fontWeight: .regular)
+        subtitleLabel.textColor = .secondaryLabel
+    }
+
+    func setupStackView() {
+        stackView.axis = .vertical
+        stackView.alignment = .leading
+        stackView.spacing = 6
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/NUX/LoginEpilogueChooseSiteTableViewCell.swift]: I think we can extract those numbers to a Constants `enum` just like was done in `LoginEpilogueCreateNewSiteCellDelegate`.

File: WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift [hunk 2]
```diff
@@ -98,7 +84,7 @@ extension LoginEpilogueTableViewController {
         }
 
         // Add one for Connect Site if there are no sites from blogDataSource.
-        if adjustedNumberOfSections == 0 && showConnectSite {
+        if adjustedNumberOfSections == 0 && showCreateNewSite {
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift:87]: Does this comment is still accurate?

File: WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift [hunk 3]
```diff
@@ -108,40 +94,64 @@ extension LoginEpilogueTableViewController {
...
+            }
         }
 
-        // Connect Site Row
-        if indexPath.row == lastRowInSection(indexPath.section) && showConnectSite {
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: LoginEpilogueConnectSiteCell.defaultReuseID) as? LoginEpilogueConnectSiteCell else {
+        // Create new site row
+        let siteRows = tableView.numberOfRows(inSection: indexPath.section)
+        let threshold = 4 // 3 site rows + 1 create site row
+        if siteRows <= threshold && indexPath.row == lastRowInSection(indexPath.section) && showCreateNewSite {
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift]: Hmm, this is doing a lot. I wonder if we can move it to a var with a more meaningful name?

**Review by leandroalonso — COMMENTED** [review]
@momo-ozawa tested Jetpack and WordPress, iPhone and iPad, light and dark mode, all good.

I've left a few suggestions to improve the code. :)

File: WordPress/Classes/ViewRelated/NUX/LoginEpilogueChooseSiteTableViewCell.swift (hunk 1 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/NUX/LoginEpilogueChooseSiteTableViewCell.swift]: Fixed! 2bb048e

File: WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift (hunk 2 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift:87]: Fixed! 32979b5

File: WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift (hunk 3 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift]: Fixed! 64fa99a

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**